packages/fiveway/src/handlers/grid.ts

```diff
@@ -14,103 +14,154 @@ export type GridItem = {
-    if (potential.row !== current.row) {
-      return null;
-    }
-    const distance = current.col - potential.col;
-    return distance > 0 ? distance : null;
-  },
-  right: (current, potential) => {
-    if (potential.row !== current.row) {
-      return null;
-    }
-    const distance = potential.col - current.col;
-    return distance > 0 ? distance : null;
-  },
+  let colDistance = potential.col - current.col;
+  if (colDistance < 0) {
+    colDistance += 0.5;
+  }
+
+  return rowDistance + Math.abs(colDistance);
 };
 
-/**
- * @category Handler
- */
-export const gridMovement: NavigationHandler = (node, action, next) => {
-  if (import.meta.env.DEV) {
-    describeHandler(action, { name: "core:grid" });
+const defaultDistanceUp = (current: GridItem, potential: GridItem) => {
+  const rowDistance = current.row - potential.row;
+  if (rowDistance <= 0) {
+    return null;
   }
 
-  if (action.kind !== "move" || action.direction === "back") {
-    return next();
+  let colDistance = current.col - potential.col;
+  if (colDistance < 0) {
+    colDistance += 0.5;
   }
 
-  const focusedId = directChildId(node.id, node.tree.focusedId);
-  if (focusedId === null) {
-    return next();
+  return rowDistance + Math.abs(colDistance);
+};
+
+const defaultDistanceLeft = (current: GridItem, potential: GridItem) => {
+  const colDistance = current.col - potential.col;
+  if (colDistance <= 0) {
+    return null;
   }
 
-  const focusedPos = GridItem.query(node.tree, focusedId);
-  if (focusedPos == null) {
-    return next();
+  let rowDistance = potential.row - current.row;
+  if (rowDistance < 0) {
+    rowDistance += 0.5;
+  }
+
+  return colDistance + Math.abs(rowDistance);
+};
+
+const defaultDistanceRight = (current: GridItem, potential: GridItem) => {
+  const colDistance = potential.col - current.col;
+  if (colDistance <= 0) {
+    return null;
+  }
+
+  let rowDistance = current.row - potential.row;
+  if (rowDistance < 0) {
+    rowDistance += 0.5;
   }
 
-  const getDistance = distanceFns[action.direction];
+  return colDistance + Math.abs(rowDistance);
+};
+
+const defaultDistance: GridHandlerConfig["distance"] = (
+  direction: NavigationDirection,
+) => {
+  switch (direction) {
+    case "up":
+      return defaultDistanceUp;
+
+    case "down":
+      return defaultDistanceDown;
+
+    case "left":
+      return defaultDistanceLeft;
+
+    case "right":
+      return defaultDistanceRight;
+  }
+};
 
-  let closestId: NodeId | null = null;
-  let shortestDistance: number | null = null;
+type GridHandlerConfig = {
+  distance?: (
+    direction: NavigationDirection,
+  ) => (a: GridItem, b: GridItem) => number | null;
+};
 
-  traverseNodes(node.tree, node.id, 1, (id) => {
-    const pos = GridItem.query(node.tree, id);
-    if (pos === null) {
-      return;
+/**
+ * @category Handler
+ */
+export const gridMovement =
+  (config: GridHandlerConfig = {}): NavigationHandler =>
+  (node, action, next) => {
```

Comment:
Could you check how `createFocusHandler` function works and implement gridMovement the same way - function with named const inside? It makes the function show up in debugger with better name. And export it just as gridMovement like in `focus.ts`